openqa-get-job-runtime-stats

```diff
@@ -0,0 +1,186 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+# 
+# Summary: Display the test runtime statistics for a range of openQA test runs
+# Job ranges can be defined via 7..9 (i,e, [7,8, 9]) or 5+2 (i.e. [5,6,7])
+# 
+# Usage example: ./openqa-get-job-runtime-stats https://openqa.opensuse.org/tests/100000+10
+# 
+
+import requests
+import datetime
+import time
+import json
+import sys
+import argparse
+import numpy as np
+
+def parse_t(dt_str) :
+	# "2023-01-25T10:22:21"
+	if dt_str is None : return None
+	dt_str = dt_str.strip()
+	if dt_str == "" : return None
+	return datetime.datetime.strptime(dt_str, '%Y-%m-%dT%H:%M:%S')
+
+
+class Job :
+	# Init job by the json object fetched from OpenQA
+	def __init__(self, obj) :
+		# Apply json values to this class. Yes, this is hacky.
+		for k in obj :
+			self.__dict__[k] = obj[k]
+	
+	def done(self) -> bool :
+		return self.state == "done"
+	def passed(self) -> bool :
+		return self.state == "done" and self.result == "passed"
+	def softfailed(self) -> bool :
+		return self.state == "done" and self.result == "softfailed"
+	def failed(self) -> bool :
+		return self.state == "done" and (self.result == "failed" or self.result == "incomplete")
+
+	def runtime(self) :
+		# Note: Timezone parsing is not needed here
+		t_finished = parse_t(self.t_finished)
+		if t_finished is None : return 0
+		t_started = parse_t(self.t_started)
+		if t_started is None : return 0
+		diff = t_finished - t_started
+		return diff.total_seconds()
+	
+	def __item__(self, index) :
+		return self._obj[index]
+
+
+def parse_job_number(jobstr) :
+	# Range? (Start..End)
+	if ".." in jobstr :
+		i = jobstr.find("..")
+		try :
+			first = int(jobstr[:i])
+			last = int(jobstr[i+2:])
+			return range(first, last+1)
+		except ValueError :
+			raise ValueError("invalid job id in job range")
+	# Sequence? (Start+Increment, e.g. 4+3 for [4,5,6,7])
+	if "+" in jobstr :
+		i = jobstr.find("+")
+		try :
+			first = int(jobstr[:i])
+			plus = int(jobstr[i+1:])
+			return range(first, first+plus+1)
+		except ValueError :
+			raise ValueError("invalid job id in job sequence")
+	
+	# Try to parse as int
+	try :
+		return [int(jobstr)]
+	except ValueError:
+		raise ValueError("invalid job identifier")
+
+def get_job_api_urls(arg, default_remote="https://openqa.opensuse.org") :
```

Comment:
This function is never called with a 2nd argument. Maybe it makes sense to make the `default_remote` actually configurable from the outside?